cobbler/actions/check.py

```diff
@@ -136,9 +138,9 @@ def check_service(self, status, which, notes=""):
             notes = " (NOTE: %s)" % notes
         return_code = 0
         if utils.is_supervisord():
-            return_code = utils.subprocess_call("supervisorctl status %s > /dev/null 2>/dev/null" % which,
-                shell=True)
-            if return_code != 0:
+            server = ServerProxy('http://localhost:9001/RPC2')
+            process_info = server.supervisor.getProcessInfo(which)
+            if process_info['statename'] != "RUNNING":
                 status.append("service %s is not running%s" % (which, notes))
                 return
```

Comment:
Suggestion from my side to prevent open connections on any problem.

```suggestion
            with ServerProxy('http://localhost:9001/RPC2') as server:
                process_info = server.supervisor.getProcessInfo(which)
                if process_info['statename'] != "RUNNING":
                    status.append("service %s is not running%s" % (which, notes))
                    return
```